benchmarks/text_hybrid_scoreboard/run_lexical_comparison.py

```
@@ -0,0 +1,277 @@
+def run_command(command: list[str], cwd: Path, timeout: int, log_path: Path, env: dict[str, str] | None = None) -> subprocess.CompletedProcess[str]:
+    merged_env = os.environ.copy()
+    if env:
+        merged_env.update(env)
+    started = time.time()
+    try:
+        result = subprocess.run(command, cwd=cwd, env=merged_env, text=True, capture_output=True, timeout=timeout, check=False)
+        text = f"command={shlex.join(command)}\ncwd={cwd}\nstarted_unix={started}\nreturncode={result.returncode}\n\n[stdout]\n{result.stdout}\n[stderr]\n{result.stderr}"
+    except subprocess.TimeoutExpired as exc:
+        text = f"command={shlex.join(command)}\ncwd={cwd}\nstarted_unix={started}\ntimeout_seconds={timeout}\n\n[stdout]\n{exc.stdout or ''}\n[stderr]\n{exc.stderr or ''}"
+        log_path.write_text(text, encoding="utf-8")
+        raise RuntimeError(f"timed out without fallback: {shlex.join(command)}") from exc
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)